ts/receiver/configMessage.ts

```diff
@@ -109,36 +114,55 @@ async function handleGroupsAndContactsFromConfigMessage(
     }
   }
   if (configMessage.contacts?.length) {
-    await Promise.all(
-      configMessage.contacts.map(async c => {
-        try {
-          if (!c.publicKey) {
-            return;
-          }
-          const contactConvo = await getConversationController().getOrCreateAndWait(
-            toHex(c.publicKey),
-            ConversationTypeEnum.PRIVATE
-          );
-          const profile: SignalService.DataMessage.ILokiProfile = {
-            displayName: c.name,
-            profilePicture: c.profilePicture,
-          };
-          // updateProfile will do a commit for us
-          contactConvo.set('active_at', _.toNumber(envelope.timestamp));
-
-          await updateProfileOneAtATime(contactConvo, profile, c.profileKey);
-        } catch (e) {
-          window?.log?.warn('failed to handle  a new closed group from configuration message');
-        }
-      })
-    );
+    await Promise.all(configMessage.contacts.map(async c => handleContactReceived(c, envelope)));
   }
 }
 
+const handleContactReceived = async (
+  contactReceived: SignalService.ConfigurationMessage.IContact,
+  envelope: EnvelopePlus
+) => {
+  try {
+    if (!contactReceived.publicKey) {
+      return;
+    }
+    const contactConvo = await getConversationController().getOrCreateAndWait(
+      toHex(contactReceived.publicKey),
+      ConversationTypeEnum.PRIVATE
+    );
+    const profile = {
+      displayName: contactReceived.name,
+      profilePictre: contactReceived.profilePicture,
+    };
+    // updateProfile will do a commit for us
+    contactConvo.set('active_at', _.toNumber(envelope.timestamp));
+
+    if (
+      window.lokiFeatureFlags.useMessageRequests &&
+      window.inboxStore?.getState().userConfig.messageRequests
+    ) {
+      if (contactReceived.isApproved) {
+        await contactConvo.setIsApproved(Boolean(contactReceived.isApproved));
+      }
+
+      if (contactReceived.isBlocked) {
+        await BlockedNumberController.block(contactConvo.id);
+      } else {
+        await BlockedNumberController.unblock(contactConvo.id);
+      }
+    }
+
+    await updateProfileOneAtATime(contactConvo, profile, contactReceived.profileKey);
```

Comment:
change this to `void` instead of `await`. just so we do not block too long the handling of messages with this cal;